src/citrine/resources/design_workflow.py

```diff
@@ -18,10 +19,34 @@ class DesignWorkflowCollection(Collection[DesignWorkflow]):
     _resource = DesignWorkflow
     _api_version = "v2"
 
-    def __init__(self, project_id: UUID, session: Session, branch_id: UUID = None):
+    def __init__(self,
+                 project_id: UUID,
+                 session: Session,
+                 branch_root_id: Optional[UUID] = None,
+                 branch_version: Optional[int] = None):
         self.project_id: UUID = project_id
         self.session: Session = session
-        self.branch_id: UUID = branch_id
+        self.branch_root_id: UUID = branch_root_id
+        self.branch_version: int = branch_version
+
+    def _resolve_branch_root_and_version(self, workflow):
+        from citrine.resources.branch import BranchCollection
```

Comment:
Since branches can also list the design workflows on them, importing this at the top of the file results in a circular dependency.